drizzle-orm/src/mysql-core/query-builders/select.types.ts

```diff
@@ -260,6 +299,61 @@ export type MySqlSelectQueryBuilder<
 
 export type AnyMySqlSelectQueryBuilder = MySqlSelectQueryBuilderBase<any, any, any, any, any, any, any, any, any>;
 
+export type AnyMySqlSetOperatorInterface = MySqlSetOperatorInterface<any, any, any, any, any, any, any, any, any>;
+
+export interface MySqlSetOperatorInterface<
```

Comment:
What is the purpose of this interface? What's the functional difference between it and `MySqlSelectBase`?

Reply:
I needed an easy way to get rid of all the private/protected methods. They were giving me a lot of problems when using the HKTs. With this helper I have access to all the types via `_`.

Reply:
I see, in that case you might simplify it with something like this I think
```ts
export type MySqlSetOperatorInterface<T extends AnyMySqlSelect> = {
	_: T['_'];
};

export type MySqlSetOperatorWithResult<TResult extends any[]> = {
	_: Omit<AnyMySqlSelect['_'], 'result'> & {
		result: TResult;
	};
};
```

Reply:
This would indeed work, but the problem is that later on, I need to use the `ValidateShape` type helper. The only way I found I could have the types behave the way I wanted, with the descriptive errors was by [narrowing it first](https://github.com/drizzle-team/drizzle-orm/pull/1218/files#diff-90b01058bcd749df48b13c9b98a75a05d6df28c40cf4b486cdc5ae99e075954bR360-R366). That's why I need this interface like this.